modin/pandas/groupby.py

```diff
@@ -837,21 +837,13 @@ def corr(self, **kwargs):
     def fillna(self, *args, **kwargs):
         new_groupby_kwargs = self._kwargs.copy()
         new_groupby_kwargs["as_index"] = True
-        work_object = type(self)(
-            df=self._df,
-            by=self._by,
-            axis=self._axis,
-            idx_name=self._idx_name,
-            drop=self._drop,
-            squeeze=self._squeeze,
-            **new_groupby_kwargs,
-        )
-        return work_object._check_index_name(
-            work_object._wrap_aggregation(
+        new_groupby_kwargs.update(kwargs)
```

Comment:
Why is `new_groupby_kwargs` getting kwargs from both `kwargs` and `self._kwargs`? I thought `self._kwargs` were the kwargs in the `groupby()` call and the param `kwargs` here were the kwargs to `fillna()`?